src/main/java/teammates/logic/core/FeedbackQuestionsLogic.java

```diff
@@ -523,51 +495,33 @@ public void updateFeedbackQuestionNumber(FeedbackQuestionAttributes newQuestion)
-            for (int i = oldQuestionNumber - 1; i >= newQuestionNumber; i--) {
-                FeedbackQuestionAttributes question = questions.get(i - 1);
-                question.questionNumber += 1;
-                try {
-                    updateFeedbackQuestionWithoutResponseRateUpdate(question);
-                } catch (InvalidParametersException e) {
-                    Assumption.fail("Invalid question.");
-                } catch (EntityDoesNotExistException e) {
-                    Assumption.fail("Question disappeared.");
-                }
-            }
-        } else if (oldQuestionNumber < newQuestionNumber && oldQuestionNumber < questions.size()) {
-            for (int i = oldQuestionNumber + 1; i <= newQuestionNumber; i++) {
-                FeedbackQuestionAttributes question = questions.get(i - 1);
-                question.questionNumber -= 1;
-                try {
-                    updateFeedbackQuestionWithoutResponseRateUpdate(question);
-                } catch (InvalidParametersException e) {
-                    Assumption.fail("Invalid question.");
-                } catch (EntityDoesNotExistException e) {
-                    Assumption.fail("Question disappeared.");
-                }
-            }
-        }
+    private void adjustQuestionNumbersAndUpdateQuestion(
+            FeedbackQuestionAttributes question,
+            int oldQuestionNumber)
+            throws InvalidParametersException, EntityDoesNotExistException, EntityAlreadyExistsException {
+        
+        fqDb.saveQuestionAndAdjustQuestionNumbers(question, true, oldQuestionNumber);
     }
-
+    
     /**
-     * Updates the feedback question. For each attribute in
-     * {@code newAttributes}, the existing value is preserved if the attribute
-     * is null (due to 'keep existing' policy). Existing responses for the
-     * question are automatically deleted if giverType/recipientType are
-     * changed, or if the response visibility is increased. However, the
-     * response rate of the feedback session is not updated.<br>
-     * Precondition: <br>
-     * {@code newAttributes} is not {@code null}
+     * Saves {@code question} to the database. Updates the question numbers of the other questions
+     * in the session based on the question number of {@code question} and {@code oldQuestionNumber}.
+     * @param question
+     * @param oldQuestionNumber
+     * @throws InvalidParametersException
+     * @throws EntityDoesNotExistException
+     * @throws EntityAlreadyExistsException
      */
-    private void updateFeedbackQuestionWithoutResponseRateUpdate(FeedbackQuestionAttributes newAttributes)
-            throws InvalidParametersException, EntityDoesNotExistException {
-
-        updateFeedbackQuestion(newAttributes, false);
+    private void adjustQuestionNumbersAndCreateQuestion(
+                FeedbackQuestionAttributes question,
+                int oldQuestionNumber)
```

Comment:
This could also be on the previous line. Also the indentation could be aligned with `throws`